core/block/import/markdown/blockconverter.go

```diff
@@ -241,40 +236,33 @@ func (m *mdConverter) createBlocksFromFile(shortPath string, f io.ReadCloser, fi
 		}
 		// md file no longer needed
 		m.processBlocks(shortPath, files[shortPath], files)
-	} else {
-		// need to store file reader, so we can use it to create local file and upload it
-		files[shortPath].ReadCloser = f
 	}
 	return nil
 }
 
-func (m *mdConverter) createFile(f *model.BlockContentFile, id string, files map[string]*FileInfo) {
-	baseName := filepath.Base(f.Name) + id
-	tempDir := m.tempDirProvider.TempDir()
-	newFile := filepath.Join(tempDir, baseName)
-	tmpFile, err := os.Create(newFile)
-	if err != nil {
-		log.Errorf("failed to create file: %s", oserror.TransformError(err).Error())
-		return
-	}
-	defer tmpFile.Close()
-	w := bufio.NewWriter(tmpFile)
+func (m *mdConverter) createFile(f *model.BlockContentFile, id string, importedSource source.Source) {
 	shortPath := f.Name
-	targetFile, found := files[shortPath]
-	if !found {
-		return
-	}
-	defer targetFile.Close()
-	_, err = w.ReadFrom(targetFile.ReadCloser)
-	if err != nil {
-		log.Errorf("failed to read file: %s", err.Error())
-		return
-	}
-
-	if err := w.Flush(); err != nil {
-		log.Errorf("failed to flush file: %s", err.Error())
+	if err := importedSource.ProcessFile(shortPath, func(fileReader io.ReadCloser) error {
+		baseName := filepath.Base(f.Name) + id
+		tempDir := m.tempDirProvider.TempDir()
+		newFile := filepath.Join(tempDir, baseName)
+		tmpFile, err := os.Create(newFile)
+		if err != nil {
+			return err
+		}
+		defer tmpFile.Close()
+		w := bufio.NewWriter(tmpFile)
+		_, err = w.ReadFrom(fileReader)
+		if err != nil {
+			return err
+		}
+		if err := w.Flush(); err != nil {
```

Comment:
It seems that you reinitialize err one more time - err is already created in line 249